src/deep-security/lib/IdentityProvider.js

```diff
@@ -96,6 +99,7 @@ export class IdentityProvider {
         userId = identityMetadata.userID;
         break;
 
+      case IdentityProvider.COGNITO_USER_POOL_PROVIDER:
       case IdentityProvider.AMAZON_PROVIDER:
       case IdentityProvider.AUTH0_PROVIDER:
         token = identityMetadata.access_token;
```

Comment:
- Add custom logic for `COGNITO_USER_POOL_PROVIDER` (@see https://github.com/aws/amazon-cognito-identity-js/blob/master/src/CognitoUserSession.js#L46 for more details)